packages/tracing/src/browser/browsertracing.ts

```diff
@@ -61,10 +61,15 @@ export interface BrowserTracingOptions extends RequestInstrumentationOptions {
   markBackgroundTransactions: boolean;
 
   /**
-   * beforeNavigate is called before a pageload/navigation transaction is created and allows for users
-   * to set custom transaction context. Default behavior is to return `window.location.pathname`.
+   * beforeNavigate is called before a pageload/navigation transaction is created and allows users to modify transaction
+   * context data, or drop the transaction entirely (by setting `sampled = false` in the context).
    *
-   * If undefined is returned, a pageload/navigation transaction will not be created.
+   * Note: For legacy reasons, transactions can also be dropped by returning `undefined`, though that option may
+   * disappear in the future.
```

Comment:
L:

```suggestion
   * Note: For legacy reasons, transactions can also be dropped by returning `undefined`.
```

We should not break the old behavior unnecessarily.

Perhaps this is not a "legacy behavior" depending on what "drop a transaction" means.

Reply:
I was thinking it would simplify things, to eventually only have one way for `beforeNavigate` to force `sampled = false` (by literally setting `sampled` to `false` in the transaction context), but I'm not wedded to it. At this point all it is is two comments in the code (one announcing the plan and one reminding us to carry it out in V6), so it's easy to pull out.